hw/ip/entropy_src/doc/_index.md

```diff
@@ -10,34 +10,38 @@ This module conforms to the [Comportable guideline for peripheral functionality.
+- This revision supports a LFSR (linear feedback shift register) digital source, and provides an interface to an external analog entropy source.
+The LFSR is used as a digital, pseudo-random type of entropy source, while the analog external source is a true entropy source.
 A noise source and its relation to an entropy source are defined by [SP 800-90B](https://csrc.nist.gov/publications/detail/sp/800-90b/final).
 - A set of registers is provided for firmware to manage and obtain entropy bits.
-- The set of registers is designed such that firmware should not have to change for the case where a random source of entropy is provided in a future revision.
+- The set of registers is designed such that firmware can select between a digital or an analog entropy source.
 - Interrupts are supported:
-  - Entropy bits are available.
-  - An internal error has occurred.
-- No SP 800-90B defined health checks are done by this revision (these functions are left for the firmware to implement).
+  - Entropy bits are available for firmware consumption.
+  - The external analog entropy source is available.
+  - The external analog entropy source has bit polarity errors.
+  - The external analog entropy source has a FIFO error.
+  - An internal FIFO error has occurred.
+- Two health checks that defined by SP 800-90B are performed by this revision:  Repetition Count and Adaptive Proportion tests.
 - No alerts are supported by this revision.
 
 
 ## Description
 
 This IP block provides an entropy source that is part of a larger solution to generate true random numbers (a TRNG mechanism).
-The general solution for TRNG relies on firmware to orchestrate the majority of the function, getting helper services from specific hardware blocks.
+The general solution for TRNG relies on either hardware or firmware to implement the majority of the function, depending which functions are enabled by firmware or e-fuses.
 For purposes of reference in this document, the NIST term "noise" will be synonymous with the term "entropy".
 Likewise, no conditioning function will be performed in this block.
-For the generation of entropy, this block provides the entropy bits needed for the entropy source solution as required by the firmware.
+For the generation of entropy, this block provides the entropy bits needed for the entropy source solution as required by the RNG mechanism.
 
-At a high-level, this ENTROPY_SRC block, when enabled, will continuously collect entropy bits from the entropy source into a FIFO that can be read from the TLUL bus.
-For this revision, the entropy generation can be very fast, based on programming.
-Other sources could be much slower.
-The support logic in the block will support entropy sources of any rate.
+At a high-level, this ENTROPY_SRC block, when enabled, will continuously collect entropy bits from the entropy source into a FIFO that can be read from the TLUL bus, or sent out through a hardware interface.
```

Comment:
@cdgori 
could you clarify this particular point a bit more also? I couldn't tell from the block diagram, if an entry from the quarantine fifos, when enabled, will spread to both HWFIFO and TLUL FIFO at the same time? Ie, will the two resultant fifos potentially end up with the same "stuff"?

Reply:
No - will never put the same entropy bits in both the TLUL and HWIF FIFOs. Later text says this:
The HWIF FIFO has lower priority than the TLUL FIFO, but is expected to have higher bandwidth demands.
Because of this, higher priority is given to the TLUL FIFO so that it does not starve when the hardware entropy interface is consuming a high level of entropy bits.

Does this explain it properly?

Reply:
As long as @mwbranstad 's comment is accurate then there is no issue to my understanding.